### DISCUSSION TITLE
When is the Elementor DOM ready within the editor?

### DISCUSSION THREAD

**Discussion by scrollingthrough** [elementor/elementor#25398] [Developers Community]
I have a few issues that I'm going through right now, the goal being to enqueue a script after Elementor's done with enqueuing its own things and then allow me to make use of the `$e` global, as well as the `elementor` one, **all of this after the Elementor DOM's loaded.**

I'm enqueueing my script as such:

```
/**
 * Let's enqueue the main JS file.
 */
add_action(
    'elementor/frontend/after_enqueue_scripts',
    [$this, 'enqueueMainScript'],
    10
);
```

It gets loaded. Inside the JS file:

```
jQuery(window).on('elementor/frontend/init', () => {
    console.log('I should have the DOM ready here...but it is not!');
    console.log('However, the elementor global is available.);
});
```

And that's when I hit the issue. Nor the dom, nor either of globals are present and ready for use. I can see that my script is loaded BEFORE Elementor's.

I don't get it. Exactly where should I `wp_enqueue_script` my script such that I have access to these things if I want to work with Elementor?

**Comment by scrollingthrough** [comment]
Good talk.

**Comment by christianswanepoel** [comment]
Hi Did you ever resolve this issue?
